src/C-DEngine/C-DCommunication/cdeQueuedSender/cde_QueuedSenderREST.cs

```diff
@@ -140,9 +148,14 @@ public void TheRESTSenderThread()
                             }
                             tQueued = GetNextMessage(out MCQCount);
                         }
+
                         if (tQueued != null)
                         {
                             TheCDEKPIs.IncrementKPI(eKPINames.QSSent);
+                            if (kpiLabels is { Count: > 0 })
```

Comment:
same here: if (EnableKPIs && kpiLabels is...

Reply:
Please see the new changes in `TheCDEKpis`.

Reply:
I would still prefer if you could enclose the new block in a "if (EnableKPIs)" because ToString and Substring() are expensive functions and will cause a performance penalty under high stress. The SenderThread is one of the most tuned functions because it is called possibly many thousand times per second in the cloud.